packages/blocks/src/store/selectors.js

```diff
@@ -95,6 +95,44 @@ export function getBlockVariations( state, blockName, scope ) {
 	} );
 }
 
+/**
+ * Returns the active block variation for a given block based on its attributes.
+ *
+ * @param {Object}                state      Data state.
+ * @param {string}                blockName  Name of block (example: “core/columns”).
+ * @param {Object}                attributes Block attributes used to determine active variation.
+ * @param {WPBlockVariationScope} [scope]    Block variation scope name.
+ *
+ * @return {(WPBlockVariation|void)} Active block variation.
+ */
+// eslint-disable-next-line camelcase
+export function unstable__getActiveBlockVariation(
+	state,
+	blockName,
+	attributes,
+	scope
+) {
+	const variations = getBlockVariations( state, blockName, scope );
+
+	const match = variations?.find( ( variation ) => {
+		if ( typeof variation.isActive === 'function' ) {
+			return variation.isActive( attributes, variation.attributes );
+		}
+
+		if ( Array.isArray( variation.isActive ) ) {
```

Comment:
We need to validate in some place the provided as strings `attributes`, because in the case of having a non existing attribute the equality results to `undefined === undefined`, which is `true` and will show the first declared variation as `active`.

Reply:
Should we bail out early when there is an empty array passed as `isActive`?